src/main/java/core/basesyntax/figures/Rectangle.java

```diff
@@ -0,0 +1,51 @@
+package core.basesyntax.figures;
+
+import core.basesyntax.suppliers.ColorSupplier;
+import java.util.Random;
+
+public class Rectangle extends Figure {
+    private final String name = "rectangle";
+    private int length;
+    private int width;
+    private final ColorSupplier colorSupplier = new ColorSupplier();
+    private final Random random = new Random();
```

Comment:
model classes cannot contain dependencies, so this is business logic and you need to move the method getRandomProperties() to FigureSuplier